### PR TITLE
Add a new process_addons task to remove AMO links in some fields

### LINKED ISSUE #9735: Remove all entries in the DB that have addons.mozilla.org as the host part of the link
See also: #9012 

---

We should remove AMO links in the following fields:

- add-on homepage
- contribution URL
- support URL

Quoting @diox:

> We need to write something to do that yes. It probably needs to be a task fired from the process_addons command.

### PR DESCRIPTION
**Pull request by willdurand**
Fixes #9735

---

As per our IRC discussion, I set the values of the translated fields to
an empty string instead of deleting the `Translation` objects.

In the test suite, I rely on the default `settings` because I cannot use
`override_settings` here (the `settings.DOMAIN` value is used in a
static `tasks` dict, which cannot be easily overridden because it is
already set when the file is imported).

### CODE REVIEW

File: src/olympia/addons/tasks.py [hunk 1]
```diff
@@ -808,3 +809,27 @@ def disable_legacy_files(ids, **kw):
...
+    """With the specified ids, remove the AMO links in the following URL fields
+    of each add-on: homepage, support_url, contribution."""
+    log.info('Deleting AMO links in URL fields %d-%d [%d].', ids[0], ids[-1],
+             len(ids))
+    addons = Addon.objects.filter(id__in=ids)
+    for addon in addons:
+        with transaction.atomic():
+            translation_ids = []
+            if addon.homepage:
+                translation_ids.append(addon.homepage.id)
```
Comment (eviljeff) [inline src/olympia/addons/tasks.py]: you can do `if addon.homepage_id:` and use `homepage_id` directly if you just need the `id`.

File: src/olympia/addons/tasks.py [hunk 2]
```diff
@@ -808,3 +809,27 @@ def disable_legacy_files(ids, **kw):
...
+            if addon.support_url:
+                translation_ids.append(addon.support_url.id)
+            if translation_ids:
+                Translation.objects.filter(
+                    id__in=translation_ids,
+                    localized_string__icontains=settings.DOMAIN
+                ).update(localized_string=u'')
```
Comment (eviljeff) [inline src/olympia/addons/tasks.py]: you should update `localized_string_clean` to `''` too (it contains the `bleach`ed, escaped, etc version of the string)

File: src/olympia/addons/tasks.py [hunk 3]
```diff
@@ -808,3 +809,27 @@ def disable_legacy_files(ids, **kw):
...
+                translation_ids.append(addon.homepage_id)
+            if addon.support_url_id:
+                translation_ids.append(addon.support_url_id)
+            if translation_ids:
+                Translation.objects.filter(
+                    id__in=translation_ids,
+                    localized_string__icontains=settings.DOMAIN
+                ).update(localized_string=u'', localized_string_clean=u'')
+            if settings.DOMAIN.lower() in addon.contributions.lower():
+                addon.update(contributions=u'')
```
[the remark this answers is not in the document]
Comment (willdurand) [reply src/olympia/addons/tasks.py]: I can probably change the condition to only remove links **starting with** `(https?://)?DOMAIN`. WDYT?

[the remark this answers is not in the document]
Comment (eviljeff) [reply src/olympia/addons/tasks.py]: I was going to suggest that but then couldn't remember if we require the `https` in the string or assume it if none provided.  Maybe just check with an sql query in redash (sql.telemetry.mozilla.org) that no addons have contributions with an AMO url in already

[the remark this answers is not in the document]
Comment (willdurand) [reply src/olympia/addons/tasks.py]: @eviljeff it looks like we do not have any entries: https://sql.telemetry.mozilla.org/queries/59744

**Pull request merged**